web-app/server/routes/me.js

```diff
@@ -336,4 +422,42 @@ router.post(
   }
 );
 
+router.get('/:subjectId/students', checkJWT, async (req, res, next) => {
+  await User.findOne({ username: req.decoded.user }, async (err, user) => {
+    if (err) throw err;
+    else {
+      const subjectId = req.params.subjectId;
+      const networkObj = await network.connectToNetwork(req.decoded.user);
+      const queryStudents = await network.query(networkObj, 'GetStudentsBySubject', subjectId);
+      const queryScore = await network.query(networkObj, 'GetScoresBySubject', subjectId);
+
+      if (!queryStudents.success || !queryScore.success) {
+        return res.json({
+          success: false,
+          msg: response.msg.toString()
+        });
+      }
+
+      let listScore = JSON.parse(queryScore.msg);
+      let listStudents = JSON.parse(queryStudents.msg);
+
+      listStudents.forEach((student) => {
```

Comment:
api này chạy 2 query `await`, lại chạy 2 vòng `for` lồng nhau thế này sẽ rất chậm
cần refactor hoặc sửa chaincode cho phù hợp để nhanh hơn trong các pull tới, pull này chưa cần sửa cũng được

Reply:
vâng bọn em sẽ sửa lại chaincode cho pull sau ạ